tianshou/policy/modelfree/sac.py

```diff
@@ -120,20 +118,20 @@ def forward(  # type: ignore
         else:
             act = dist.rsample()
         log_prob = dist.log_prob(act).unsqueeze(-1)
-        if self.action_bound_method == "tanh" and self.action_space is not None:
-            # apply correction for Tanh squashing when computing logprob from Gaussian
-            # You can check out the original SAC paper (arXiv 1801.01290): Eq 21.
-            # in appendix C to get some understanding of this equation.
-            if self.action_scaling:
-                action_scale = to_torch_as(
-                    (self.action_space.high - self.action_space.low) / 2.0, act)
-            else:
-                action_scale = 1.0  # type: ignore
-            squashed_action = torch.tanh(act)
-            log_prob = log_prob - torch.log(
-                action_scale * (1 - squashed_action.pow(2)) + self.__eps
-            ).sum(-1, keepdim=True)
-        return Batch(logits=logits, act=act, state=h, dist=dist, log_prob=log_prob)
+        # apply correction for Tanh squashing when computing logprob from Gaussian
+        # You can check out the original SAC paper (arXiv 1801.01290): Eq 21.
+        # in appendix C to get some understanding of this equation.
+        if self.action_scaling and self.action_space is not None:
+            action_scale = to_torch_as(
+                (self.action_space.high - self.action_space.low) / 2.0, act)
+        else:
+            action_scale = 1.0  # type: ignore
+        squashed_action = torch.tanh(act)
+        log_prob = log_prob - torch.log(
+            action_scale * (1 - squashed_action.pow(2)) + self.__eps
+        ).sum(-1, keepdim=True)
+        return Batch(logits=logits, act=squashed_action,
```

Comment:
`act=squashed_action` will cause error in larger action space since you abandon the map_action function:
```python
In [1]: import gym

In [2]: e=gym.make("Pendulum-v0")

In [3]: e.action_space.high
Out[3]: array([2.], dtype=float32)

In [4]: e.action_space.low
Out[4]: array([-2.], dtype=float32)
```

Reply:
In sac, this is solved by action scaling in map action. But for imitation policy, I think you are right.
`    il_policy.map_action = policy.map_action`
is indeed necessary.